extensions/amp-viewer-integration/0.1/messaging.js

```diff
@@ -19,9 +19,29 @@ import {listen} from '../../../src/event-helper';
 import {dev} from '../../../src/log';
 
 const TAG = 'amp-viewer-messaging';
-const sentinel_ = '__AMPHTML__';
-const requestSentinel_ = sentinel_ + 'REQUEST';
-const responseSentinel_ = sentinel_ + 'RESPONSE';
+const APP = '__AMPHTML__';
+
+/**
+ * @enum {string}
+ * @private
+ */
+const MessageType_ = {
+  REQUEST: 'q',
+  RESPONSE: 's',
+};
+
+/**
+ * @typedef {{
+ *   app: !string,
```

Comment:
Primitives (string, number, boolean) do not need `!` - they are non-nullable by default.